repair/repair.hh

```diff
@@ -507,10 +516,29 @@ struct node_ops_cmd_request {
     std::list<gms::inet_address> leaving_nodes;
     // Map existing nodes to replacing nodes
     std::unordered_map<gms::inet_address, gms::inet_address> replace_nodes;
+    std::unordered_map<gms::inet_address, std::list<dht::token>> bootstrap_nodes;
+    node_ops_cmd_request(node_ops_cmd command,
+            utils::UUID uuid,
+            std::list<gms::inet_address> ignore = {},
+            std::list<gms::inet_address> leaving = {},
+            std::unordered_map<gms::inet_address, gms::inet_address> replace ={},
+            std::unordered_map<gms::inet_address, std::list<dht::token>> bootstrap= {})
+        : cmd(command)
+        , ops_uuid(std::move(uuid))
+        , ignore_nodes(std::move(ignore))
+        , leaving_nodes(std::move(leaving))
+        , replace_nodes(std::move(replace))
+        , bootstrap_nodes(std::move(bootstrap)) {
```

Comment:
what is the real gain by adding noexcept here.

Reply:
The problem is do we really care about the exception safeness for this class? If not, why do we bother to type noexcept everywhere. Keep simple thing simple. IMHO, it is better not to use it by default.

Reply:
what's complicated with adding noexcept where it is obvious?
similarly, why do we care about `const`? we could do without it too...

Reply:
Sometimes maybe it is obvious and sometimes it is not. We use use a c++ feature not because it is obvious. Adding noexcept here will cause the whole scylla process to terminate if we the function actually throws.  In no ways, such user operations should cause a scylla process to terminate.

Reply:
> Adding noexcept here will cause the whole scylla process to terminate if we the function actually throws.
But it doesn't.
Maybe this case is too high level, but it's very important for lower level api's to be marked noexcept (when they are) so that their users can rely on that.  The boost library, for example, rarely uses `noexcept`, unlike the standard c++ library and it got us into trouble more than once when, for example, we require lambda functions to be nothrow move constructible and they capture a boost container which move constructor is not marked noexcept.